src/kubernetes/client/config/in_cluster.rs

```diff
@@ -0,0 +1,68 @@
+//! Everything related to building in-cluster configuration.
+
+use super::Config;
+use crate::tls::TlsOptions;
+use http02::Uri;
+use snafu::{ResultExt, Snafu};
+
+impl Config {
+    /// Prepares a config sutable for use when running in k8s cluster.
+    pub fn in_cluster() -> Result<Self, Error> {
+        let host = std::env::var("KUBERNETES_SERVICE_HOST").context(NotInCluster)?;
```

Comment:
Is the environment really where we should be keeping this? Can the config not hold this?

Reply:
This is how kubernetes specifies the in-cluster API access. We could put it in config, but it's not much value, since it's never going to be changes, and would always be set to env var resolution (i.e. `myval: $KUBERNETES_SERVICE_HOST`).
So, the tldr is it doesn't make sense.

Reply:
Okay it's fine. :) As long as we're not constantly polling env or something.